### PR TITLE
add observed to var_to_name Dict

### PR DESCRIPTION
**Pull request by isaacsas**
Closes https://github.com/SciML/ModelingToolkit.jl/pull/1343

### CODE REVIEW

File: src/systems/diffeqs/odesystem.jl [hunk 1]
```diff
@@ -137,6 +137,7 @@ function ODESystem(
     var_to_name = Dict()
     process_variables!(var_to_name, defaults, dvs′)
     process_variables!(var_to_name, defaults, ps′)
+    !isempty(observed) && collect_var_to_name!(var_to_name, (eq.lhs for eq in observed))
```
Comment (YingboMa) [inline src/systems/diffeqs/odesystem.jl:140]:
```suggestion
    isempty(observed) || collect_var_to_name!(var_to_name, (eq.lhs for eq in observed))
```

**Comment by YingboMa** [comment]
Do you have a use case for this? `observed` is not intended to be used by users.

**Comment by isaacsas** [comment]
Certain bio file formats include "groups" which are observables of the states. Currently the ability to `@unpack` them does not propagate when converting `ReactionSystem`s to other system types (i.e. they don't unpack as the variable but as the full equation if I recall right).

**Comment by isaacsas** [comment]
Examples include SBML, Antimony and BioNetGen.

**Comment by YingboMa** [comment]
This is a breaking change, but sure, we can do that.

**Comment by isaacsas** [comment]
OK, I can finish off the PR and leave it to merge for whenever the next breaking release is planned. I think it makes sense; one can unpack parameters and states, so it seems like one should be able to unpack observable variables too to work with them.

**Comment by isaacsas** [comment]
@YingboMa I think this is good to go unless you have any other comments. Thanks!

**Review by YingboMa — APPROVED** [review]
LGTM

**Comment by YingboMa** [comment]
Let's merge this in this next breaking release.

**Comment by YingboMa** [comment]
Ah, merge error.

File: test/discretesystem.jl [hunk 2]
```diff
@@ -112,3 +112,22 @@ linearized_eqs = [
     y(t - 2.0) ~ y(t)
 ]
 @test all(eqs2 .== linearized_eqs)
+
```
Comment (YingboMa) [inline test/discretesystem.jl:115]: @isaacsas Sorry, this is a merge error. Could you delete this?

Comment (isaacsas) [reply test/discretesystem.jl:115]: You mean to delete the extra test code the merge added? Sure, I’ll do that tonight when I have a bit of time free.

**Comment by YingboMa** [comment]
Yes. Thanks!

**Comment by isaacsas** [comment]
@YingboMa all removed. Should be good if tests pass.

**Pull request merged**